### ISSUE TITLE
ICP (iterative closest point registration) method for point clouds with no mesh structure

### ISSUE THREAD

**Issue by bing-jian** [libigl/libigl#1588]
#### Describe your question

Does ICP in libigl work with point clouds with no mesh structure?  According to
https://github.com/libigl/libigl/blob/3cb4894eaf8ea4610467189ca292be349425d44b/include/igl/iterative_closest_point.h
seems the current implementation requires faces.  

Thanks,
Bing

#### Check all that apply (change to `[x]`)
- [ x ] Windows
- [ x ] macOS
- [ x ] Linux

**Comment by jdumas** [comment]
Yes, the code samples points uniformly on the mesh surface to perform the rigid alignment. I believe you can iteratively call `igl::rigid_alignment` on its own though. Please also check out this [recent paper](https://gfx.cs.princeton.edu/pubs/Rusinkiewicz_2019_ASO/index.php) on symmetric ICP.

**Comment by bing-jian** [comment]
Thanks for the quick response.  So I guess there is no such read-to-use function interface by now and one has to write their own version similar to 
https://github.com/libigl/libigl/blob/3cb4894eaf8ea4610467189ca292be349425d44b/include/igl/iterative_closest_point.cpp
